x/participation/keeper/available_allocations_test.go

```diff
@@ -0,0 +1,76 @@
+package keeper_test
+
+import (
+	"strconv"
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+
+	testkeeper "github.com/tendermint/spn/testutil/keeper"
+	"github.com/tendermint/spn/testutil/sample"
+	"github.com/tendermint/spn/x/participation/types"
+)
+
+func TestAvailableAllocationsGet(t *testing.T) {
+	sdkCtx, tk, _ := testkeeper.NewTestSetup(t)
+
+	invalidAddress := strconv.Itoa(1)
+	allocationPrice := types.AllocationPrice{Bonded: sdk.NewInt(100)}
+
+	tk.ParticipationKeeper.SetParams(sdkCtx, types.Params{
+		AllocationPrice: allocationPrice,
+	})
+
+	validAddress := sample.Address()
+	validAddressNoUse := sample.Address()
+	addressNegativeDelegations := sample.Address()
+
+	tk.DelegateN(sdkCtx, validAddress, 100, 10)
+	tk.DelegateN(sdkCtx, validAddressNoUse, 100, 10)
+	tk.DelegateN(sdkCtx, addressNegativeDelegations, -100, 10)
```

Comment:
Is a negative delegation possible by design?

Reply:
No, not directly through the standard mechanisms of the `staking` module.  This would only be a protection against another module that uses the `staking` keeper to create some kind of incorrect state.